src/com/facebook/buck/apple/UmbrellaHeaderGenerator.java

```diff
@@ -0,0 +1,33 @@
+package com.facebook.buck.apple;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+
+import org.stringtemplate.v4.ST;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Creates an umbrella header from a list of paths
+ */
+public class UmbrellaHeaderGenerator {
+  private static final String UMBRELLA_HEADER_TEMPLATE = "umbrella_header.st";
+
+  public static String generateUmbrellaHeader(String frameworkName, Iterable<Path> headers) {
```

Comment:
This is going to be used as part of umbrella header generation for building frameworks with buck too, so belongs in a separate class.